app/logic/Mail/ActiveSync/ActiveSyncFolder.ts

```diff
@@ -98,36 +98,40 @@ export class ActiveSyncFolder extends Folder implements ActiveSyncPingable {
       // If the function currently holding the sync key throws, we don't care.
     }
     try {
-      this.syncKeyBusy = this.makeSyncRequest(data);
+      this.syncKeyBusy = this.makeSyncRequest(data, callback);
       return await this.syncKeyBusy;
     } finally {
       this.syncKeyBusy = null;
     }
   }
 
-  protected async makeSyncRequest(data?: any): Promise<any> {
-    let request = {
-      Collections: {
-        Collection: Object.assign({
-          SyncKey: this.syncState || "0",
-          CollectionId: this.id,
-        }, data),
-      },
-    };
-    let response = await this.account.callEAS("Sync", request);
-    if (!response) {
-      return null;
-    }
-    if (response.Collections.Collection.Status == "3") {
-      // Out of sync.
-      this.syncState = null;
+  protected async makeSyncRequest(data?: any, callback?: (response: any) => Promise<void>): Promise<any> {
```

Comment:
Please document `responseFunc`: what it does, when it's called, and what is the parameter value.
Important: Also document that it may be called many times, which is very important for the caller and its flow.